File: frontend/lib/core/models/user_role.dart
/// Mirrors the backend's App\Enums\UserRole (backend CLAUDE.md rule 11).
enum UserRole {
  superAdmin('SUPER_ADMIN', 'Super Admin'),

  /// Reads every branch in a school group and writes into whichever it
  /// names. Not a platform role - see docs/branches.md.
  groupAdmin('GROUP_ADMIN', 'Group Admin'),
  schoolAdmin('SCHOOL_ADMIN', 'School Admin'),
  hod('HOD', 'HOD'),
  teacher('TEACHER', 'Teacher'),
  staff('STAFF', 'Staff'),
  transportManager('TRANSPORT_MANAGER', 'Transport Manager'),

  /// Runs payroll for their own school; otherwise an ordinary employee.
  /// See docs/payroll.md.
  accountant('ACCOUNTANT', 'Accountant');

  const UserRole(this.apiValue, this.label);

  final String apiValue;
  final String label;

  /// Whether a form has to ask which school is deliberately *not* here: a
  /// School Admin spans a group or a single school depending on the school
  /// they are in, which a role cannot know. See
  /// AuthenticatedUser.picksSchool.

  /// Administers schools' own affairs - their own school, and every branch in
  /// its group where there is one. Mirrors the backend's
  /// UserRole::administersSchool().
  bool get administersSchool => this == UserRole.schoolAdmin || this == UserRole.groupAdmin;

  static UserRole fromApiValue(String value) {
    return UserRole.values.firstWhere(
      (role) => role.apiValue == value,
      orElse: () => throw ArgumentError('Unknown role: $value'),
    );
  }
}

File: backend/app/Enums/UserRole.php
<?php

namespace App\Enums;

enum UserRole: string
{
    case SuperAdmin = 'SUPER_ADMIN';
    // Attached to a parent school: reads every branch in its group, and
    // writes into whichever branch it names. Not a platform role - schools
    // and payments stay SUPER_ADMIN. See docs/branches.md.
    case GroupAdmin = 'GROUP_ADMIN';
    // Attached to any one school. Where that school is part of a group, the
    // same group reach as a Group Admin - the head office looks down at its
    // branches and a branch looks up and across at its sisters. Where it is
    // standalone, which is most schools, exactly one school as always.
    case SchoolAdmin = 'SCHOOL_ADMIN';
    case Hod = 'HOD';
    case Teacher = 'TEACHER';
    case Staff = 'STAFF';
    case TransportManager = 'TRANSPORT_MANAGER';

    /**
     * Administers schools' own affairs - their own school, and every branch
     * in its group where there is one.
     *
     * Says nothing about *which* schools: that is SchoolScope's question, and
     * asking it here instead is how the two would drift apart.
     *
     * Deliberately not "is an admin": onboarding a school and recording the
     * payments it makes are platform actions and stay SUPER_ADMIN.
     */
    public function administersSchool(): bool
    {
        return $this === self::SchoolAdmin || $this === self::GroupAdmin;
    }
}
